batchspawner/batchspawner.py

```diff
@@ -540,6 +554,25 @@ class SlurmSpawner(UserEnvMixin,BatchSpawnerRegexStates):
 echo "jupyterhub-singleuser ended gracefully"
 {{epilogue}}
 """).tag(config=True)
+
+    # all these req_foo traits will be available as substvars for templated strings
+    req_cluster = Unicode('',
+        help="Cluster name to submit job to resource manager"
+        ).tag(config=True)
+
+    req_qos = Unicode('',
+        help="QoS name to submit job to resource manager"
+        ).tag(config=True)
+
+    req_srun = Unicode('srun',
+        help="Job step wrapper, default 'srun'.  Set to '' you do not want "
```

Comment:
Perhaps: Set `req_srun=''` to disable running in job step, and note that this affects environment handling.

Reply:
Done, thanks.